api/v1/clusterextensionrevision_types.go

```diff
@@ -55,9 +55,9 @@ type ClusterExtensionRevisionSpec struct {
 	// a unique value assigned. It is immutable after creation. The new revision number must always be previous revision +1.
 	//
 	// +kubebuilder:validation:Required
-	// +kubebuilder:validation:Minimum:=1
+	// +kubebuilder:validation:Minimum=1
 	// +kubebuilder:validation:XValidation:rule="self == oldSelf", message="revision is immutable"
-	Revision int64 `json:"revision"`
+	Revision int32 `json:"revision"`
```

Comment:
what we are improving/fixing with changing the type?

Reply:
It seems to be best practice to use int32 if the extra precision isn't needed as it avoid issues with JSON serialization. Which unless we expect to go over 2.1B revisions, we won't see. 

I don't have a big dog in this hunt since I don't think we'll experience the json serialization issues (since we'll never reach a number large enough for it to show up). I guess we could also cap the value at the JSON max int to avoid this? I'd also like to understand whether we think we need the extra precision and what do we lose by not having it...

It would be a breaking change. But, is my understanding correct that we are allowed to have breaking API changes while in experimental mode? Or is the expectation that we change the version? e.g. v1alpha2?